collect_stata/write_json.py

```diff
@@ -0,0 +1,369 @@
+            "frequencies": frequencies,
+        }
+    )
+    dataframe["labels"] = dataframe["labels"].astype(str)
+    dataframe["values"] = pd.to_numeric(dataframe["values"])
+    dataframe.sort_values(by="values", inplace=True)
+    return dataframe.to_dict("list")
+
+
+def uni_cat(elem, file_csv):
+    """Generate dict with frequencies and labels for categorical variables
+
+    Input:
+    elem: dict
+    file_csv: pandas DataFrame
+
+    Output:
+    cat_dict: dict
+    """
+
+    frequencies = []
+    values = []
+    missings = []
+    labels = []
+
+    stata_missings = {
+        "4294967287": "-9",
+        "4294967288": "-8",
+        "4294967289": "-7",
+        "4294967290": "-6",
+        "4294967291": "-5",
+        "4294967292": "-4",
+        "4294967293": "-3",
+        "4294967294": "-2",
+        "4294967295": "-1",
+    }
+
+    value_count = file_csv[elem["name"]].value_counts()
+    for value in elem["values"]:
+        try:
+            frequencies.append(int(value_count[value["value"]]))
+        except KeyError:
+            frequencies.append(0)
+        labels.append(value["label"])
+
+        var_value = str(value["value"])
+
+        if int(value["value"]) >= 0 and var_value not in stata_missings:
+            missings.append(False)
+            values.append(var_value)
+        else:
+            missings.append(True)
+            if var_value in stata_missings:
+                values.append(stata_missings[var_value])
+            else:
+                values.append(var_value)
+
+    cat_dict = sorting_dataframe(values, labels, missings, frequencies)
+
+    return cat_dict
+
+
+def uni_string():
+    """Generate dict with frequencies for nominal variables
+
+    Output:
+    string_dict: OrderedDict
+    """
+
+    string_dict = OrderedDict()
+
+    string_dict["frequencies"] = []
+    string_dict["labels"] = []
+    string_dict["missings"] = []
+    string_dict["values"] = []
+
+    return string_dict
+
+
+def uni_number():
+    """Generate dict with frequencies for numerical variables
+
+    Output:
+    number_dict: OrderedDict
+    """
+    number_dict = OrderedDict()
+
+    number_dict["frequencies"] = []
+    number_dict["labels"] = []
+    number_dict["missings"] = []
+    number_dict["values"] = []
+    number_dict["labels_de"] = []
+
+    return number_dict
+
+
+def stats_cat(elem, file_csv):
+    """Generate dict with statistics for categorical variables
+
+    Input:
+    elem: dict
+    file_csv: pandas DataFrame
+
+    Output:
+    statistics: OrderedDict
+    """
+
+    names = ["valid", "invalid"]
+    values = []
+
+    total = int(file_csv[elem["name"]].size)
+    invalid = int(file_csv[elem["name"]].isnull().sum()) + int(
+        sum(n < 0 for n in file_csv[elem["name"]])
+    )
+    valid = total - invalid
+
+    value_names = [valid, invalid]
+
+    for value_name in value_names:
+        values.append(str(value_name))
+
+    statistics = OrderedDict([("names", names), ("values", values)])
```

Comment:
Please return a dictionary containing the pairs {valid: value, invalid: value} directly. No lists,

Reply:
changed it for the functions stats_cat and stats_string

Reply:
Are there more stats-functions?